src/Storages/StorageDistributed.cpp

```diff
@@ -878,7 +878,7 @@ void StorageDistributed::renameOnDisk(const String & new_path_to_table_data)
     {
         const String path(disk->getPath());
         auto new_path = path + new_path_to_table_data;
-        Poco::File(path + relative_data_path).renameTo(new_path);
+        disk->moveDirectory(path + relative_data_path, new_path);
```

Comment:
You don't need to pass disk->path to disk methods explicitly. Every disk method already adds the disk path to paths given in parameters.

Reply:
Ok.